crates/sui-framework/packages/sui-framework/sources/random.move

```diff
@@ -306,4 +306,21 @@ module sui::random {
     public fun generator_buffer(r: &RandomGenerator): &vector<u8> {
         &r.buffer
     }
+
+    #[test_only]
+    /// Random generator from a non-deterministic seed.
+    /// To be used when non-deterministic randomness is needed in tests (e.g., fuzzing).
+    public fun new_generator_for_testing(): RandomGenerator {
+        let seed = generate_rand_seed();
+        new_generator_from_seed(seed)
+    }
+
+    #[test_only]
+    /// Random generator from a given seed.
+    public fun new_generator_from_seed(seed: vector<u8>): RandomGenerator {
+        RandomGenerator { seed, counter: 0, buffer: vector[] }
+    }
+
+    #[test_only]
+    native fun generate_rand_seed(): vector<u8>;
```

Comment:
Could we give this a scarier name (include `for_testing` or similar) to make sure no one accidentally removes the `test_only`?

Reply:
renamed in https://github.com/MystenLabs/sui/pull/17225